### PR TITLE
[PM-12406] Introduce new endpoint and replace SSO details response flow

### PR DESCRIPTION
**Pull request by andrebispo5**
## 🎟️ Tracking

[[PM-12406] Introduce new endpoint and replace SSO details response flow](https://bitwarden.atlassian.net/browse/PM-12406)

## 📔 Objective

In order to align the back-end contract with the front-end expectations, add a net-new GET verified domains flow.
We will keep the original SSO details flow in place, this new implementation will be behind a remote feature flag.

## 📸 Screenshots

https://github.com/user-attachments/assets/8ce7753e-2535-4ad1-8cdb-aa37270c0635

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-12406]: https://bitwarden.atlassian.net/browse/PM-12406?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -1073,4 +1073,5 @@ Do you want to switch to this account?</string>
   <string name="bitwarden_tools">Bitwarden Tools</string>
   <string name="got_it">Got it</string>
   <string name="no_logins_were_imported">No logins were imported</string>
+  <string name="verified_sso_domain_verified">Verified Sso Domain Endpoint</string>
```
Comment (david-livefront) [inline app/src/main/res/values/strings.xml:1076]: Should Sso be all caps?

File: app/src/test/java/com/x8bit/bitwarden/data/auth/datasource/network/service/OrganizationServiceTest.kt [hunk 2]
```diff
@@ -103,6 +111,26 @@ class OrganizationServiceTest : BaseServiceTest() {
         val result = organizationService.getOrganizationKeys("orgId")
         assertTrue(result.isFailure)
     }
+
+    @Suppress("MaxLineLength")
+    @Test
+    fun `getVerifiedOrganizationDomainSsoDetails when response is success should return valid response`() = runTest {
```
Comment (david-livefront) [inline app/src/test/java/com/x8bit/bitwarden/data/auth/datasource/network/service/OrganizationServiceTest.kt:117]: The formatter should move the `runTest {` down a line if it's past the line limit

File: app/src/test/java/com/x8bit/bitwarden/data/auth/datasource/network/service/OrganizationServiceTest.kt [hunk 3]
```diff
@@ -103,6 +111,27 @@ class OrganizationServiceTest : BaseServiceTest() {
         val result = organizationService.getOrganizationKeys("orgId")
         assertTrue(result.isFailure)
     }
+
+    @Suppress("MaxLineLength")
+    @Test
+    fun `getVerifiedOrganizationDomainSsoDetails when response is success should return valid response`() =
+    runTest {
```
Comment (david-livefront) [inline app/src/test/java/com/x8bit/bitwarden/data/auth/datasource/network/service/OrganizationServiceTest.kt:118]: You should re-run the formatter here, all the indentation is off

Comment (andrebispo5) [reply app/src/test/java/com/x8bit/bitwarden/data/auth/datasource/network/service/OrganizationServiceTest.kt]: Sorry thought I had it running automatically. Fixed now. 👍

**Review by david-livefront — APPROVED** [review]

**Pull request merged**